### ISSUE TITLE
Akka http client for kinesis connector not sending cloud watch metrics properly

### ISSUE THREAD

**Issue by cjaber** [akka/alpakka#2399]
### Versions used 

Akka version:  "2.6.8"
Akka http version:  "10.1.11"
Alpakka kinesis version: "2.0.1"

### Expected Behavior

When using the akka http as a http client for the configuration of the AWS clients (Dynamodb, cloudwatch , kinesis)  in the context of the configuration of a AWS KCL Scheduler Source (https://doc.akka.io/docs/alpakka/current/kinesis.html#aws-kcl-scheduler-source-checkpointer) I expected cloud watch client to being able to push metrics

### Actual Behavior

Cloud watch client is unable to push metrics with the following exception. The use of a **NettyNioAsyncHttpClient** instead of the Akka http client solves the problem

```
22:45:56.948` WARN  k.m.CloudWatchMetricsPublisher Could not publish 16 datums to CloudWatch
software.amazon.awssdk.services.cloudwatch.model.CloudWatchException: When Content-Type:application/x-www-form-urlencoded, URL cannot include query-string parameters (after '?'): '/?Action=PutMetricData&Version=2010-08-01&Namespace=
....
=1.0&MetricData.member.16.Unit=Count' (Service: CloudWatch, Status Code: 400, Request ID: eb8768c0-07ee-475b-8ef1-5a0f7d288f88, Extended Request ID: null)
        at software.amazon.awssdk.services.cloudwatch.model.CloudWatchException$BuilderImpl.build(CloudWatchException.java:95)
        at software.amazon.awssdk.services.cloudwatch.model.CloudWatchException$BuilderImpl.build(CloudWatchException.java:55)
        at software.amazon.awssdk.protocols.query.internal.unmarshall.AwsXmlErrorUnmarshaller.unmarshall(AwsXmlErrorUnmarshaller.java:99)
        at software.amazon.awssdk.protocols.query.unmarshall.AwsXmlErrorProtocolUnmarshaller.handle(AwsXmlErrorProtocolUnmarshaller.java:102)
        at software.amazon.awssdk.protocols.query.unmarshall.AwsXmlErrorProtocolUnmarshaller.handle(AwsXmlErrorProtocolUnmarshaller.java:82)
        at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.lambda$handle$0(MetricCollectingHttpResponseHandler.java:52)
        at software.amazon.awssdk.core.internal.util.MetricUtils.measureDurationUnsafe(MetricUtils.java:64)
        at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.handle(MetricCollectingHttpResponseHandler.java:52)
        at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler.lambda$prepare$0(AsyncResponseHandler.java:88)
        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
        at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler$BaosSubscriber.onComplete(AsyncResponseHandler.java:129)
        at akka.stream.impl.ReactiveStreamsCompliance$.tryOnComplete(ReactiveStreamsCompliance.scala:114)
        at akka.stream.impl.fusing.ActorGraphInterpreter$ActorOutputBoundary.complete(ActorGraphInterpreter.scala:390)
        at akka.stream.impl.fusing.ActorGraphInterpreter$ActorOutputBoundary.onUpstreamFinish(ActorGraphInterpreter.scala:416)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:523)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:625)
        at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute(ActorGraphInterpreter.scala:56)
        at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute$(ActorGraphInterpreter.scala:52)
        at akka.stream.impl.fusing.ActorGraphInterpreter$RequestMore.execute(ActorGraphInterpreter.scala:274)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:600)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:769)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:784)
        at akka.actor.Actor.aroundReceive(Actor.scala:537)
        at akka.actor.Actor.aroundReceive$(Actor.scala:535)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:691)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:577)
        at akka.actor.ActorCell.invoke(ActorCell.scala:547)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

### Relevant logs

see previous section

### Reproducible Test Case
use the following sources as configured in the following code
```
// AWS creds
  lazy val awsCreds = DefaultCredentialsProvider.create()

  //-----------------------------------------------------------------------
  //                           AWS Services Clients
  //-----------------------------------------------------------------------
  //val httpClient: SdkAsyncHttpClient = NettyNioAsyncHttpClient.builder().maxConcurrency(100).build()
  val httpClient = AkkaHttpClient.builder().withActorSystem(system).build()


  // Kinesis client
  implicit val kinesisClient: KinesisAsyncClient = {
    val builder = KinesisAsyncClient.builder()
    builder.httpClient(httpClient)
    settings.awsRegion.foreach(region => Region.of(region))
    builder.credentialsProvider(awsCreds)
    builder.build()
  }

  // DynamoDB client
  implicit val dynamoClient: DynamoDbAsyncClient = {
    val builder = DynamoDbAsyncClient.builder()
    builder.httpClient(httpClient)
    settings.awsRegion.foreach(region => Region.of(region))
    builder.credentialsProvider(awsCreds)
    builder.build()
  }


  // Dynamo client
  implicit val cloudWatchClient: CloudWatchAsyncClient = {
    val builder = CloudWatchAsyncClient.builder()
    builder.httpClient(httpClient)
    settings.awsRegion.foreach(region => Region.of(region))
    builder.credentialsProvider(awsCreds)
    builder.build()
  }

  system.registerOnTermination(kinesisClient.close())
  system.registerOnTermination(dynamoClient.close())
  system.registerOnTermination(cloudWatchClient.close())


  //-----------------------------------------------------------------------
  //                           Akka Stream Kinesis Source/Sink Config
  //-----------------------------------------------------------------------

  private val schedulerSourceSettings = KinesisSchedulerSourceSettings(bufferSize = 1000, backpressureTimeout = 1.minute)
  private val checkpointSettings = KinesisSchedulerCheckpointSettings(100, 30.seconds)

  private val sourceBuilder: ShardRecordProcessorFactory => Scheduler =
    recordProcessorFactory => {

      val configsBuilder = new ConfigsBuilder(
        appOutputStreamName,
        s"agg-translator-app",
        kinesisClient,
        dynamoClient,
        cloudWatchClient,
        instanceIp + "#" + instanceIp,
        recordProcessorFactory
      )

      new Scheduler(
        configsBuilder.checkpointConfig,
        configsBuilder.coordinatorConfig,
        configsBuilder.leaseManagementConfig,
        configsBuilder.lifecycleConfig,
        configsBuilder.metricsConfig,
        configsBuilder.processorConfig,
        //Disable enhanced out mode for kinesis
        configsBuilder.retrievalConfig().retrievalSpecificConfig(new PollingConfig(appOutputStreamName, kinesisClient))
      )
    }

  val aggregatedSource = KinesisSchedulerSource(sourceBuilder, schedulerSourceSettings)
  val checkpointSink = KinesisSchedulerSource.checkpointRecordsSink(checkpointSettings)
```

**Comment by ennru** [comment]
Thank you for reporting, I assume this is a problem whenever using https://github.com/matsluni/aws-spi-akka-http to enable Akka HTTP as the AWS client.

@matsluni do you have any insights to share?

**Comment by matsluni** [comment]
Thanks for mentioning me here. It looks like it's related to this one https://github.com/matsluni/aws-spi-akka-http/issues/55.

I can have a closer look at this. But I am currently on vacation and can look at this earliest next week.

**Comment by matsluni** [comment]
I had a look at this, but unfortunately was not able to reproduce it. Also the code above is a bit incomplete.

@cjaber is it possible for you to create a small self contained example where the error is consistently reproducible?

**Comment by matsluni** [comment]
I created a draft branch which has a fallback to an empty Content Type if none is supplied in the original request. As I could not reproduce the error I am not 100% sure this solves the error you see. 

https://github.com/matsluni/aws-spi-akka-http/tree/Empty_ContentType_allowed

**Comment by matsluni** [comment]
I just released v0.0.10 of the library with the fallback `ContentTypes.NoContentType`. @cjaber, can you please verify, that this change solves your problem.

**Comment by cjaber** [comment]
Hello @matsluni  and sorry for the delay!!  In my project your lib is a transitive dependency of 'akka-stream-alpakka-kinesis' nevertheless I managed to use the last version of your lib 0.0.10 and seems to work fine. Hopefully this version will be used in the next akka stream connectors versions to be able to use it properly!

Thanks again!!

**Comment by ennru** [comment]
Thank you for verifying that it solves this issue!
I've just opened https://github.com/akka/alpakka/pull/2432 to bring the update to Alpakka.

**Comment by matsluni** [comment]
@cjaber, thanks for verifying the fix and thanks @ennru for creating the PR in Alpakka to update to the new version.

**Comment by ennru** [comment]
Fixed with #2432. Until the next Alpakka release becomes available, users may pull up the dependency themselves.
